pkg/controllermanager/controller/shoot/shoot_maintenance_test.go

```diff
@@ -247,24 +283,84 @@ var _ = Describe("Shoot Maintenance", func() {
 				},
 			}
 		})
-		It("should determine that the shoot kubernetes version must be maintained - ForceUpdate", func() {
+		It("should determine that the shoot kubernetes version must be maintained - ForceUpdate to latest patch version", func() {
 			shoot.Spec.Maintenance.AutoUpdate.KubernetesVersion = false
-			cloudProfile.Spec.Kubernetes.Versions[2].ExpirationDate = &expirationDateInThePast
-			shoot.Spec.Kubernetes = gardencorev1beta1.Kubernetes{Version: cloudProfile.Spec.Kubernetes.Versions[2].Version}
+			cloudProfile.Spec.Kubernetes.Versions[4].ExpirationDate = &expirationDateInThePast
+			shoot.Spec.Kubernetes = gardencorev1beta1.Kubernetes{Version: "1.0.1"}
 
-			version, err := MaintainKubernetesVersion(shoot, cloudProfile)
+			version, _, err := MaintainKubernetesVersion(shoot, cloudProfile, shootLogger)
 
 			Expect(err).To(BeNil())
 			Expect(version).NotTo(BeNil())
 			Expect(*version).To(Equal("1.0.2"))
 		})
 
+		It("should determine that the shoot kubernetes version must be maintained - ForceUpdate to latest non-preview patch version", func() {
+			shoot.Spec.Maintenance.AutoUpdate.KubernetesVersion = false
+			// expire shoots kubernetes version 1.0.0
+			cloudProfile.Spec.Kubernetes.Versions[5].ExpirationDate = &expirationDateInThePast
+			shoot.Spec.Kubernetes = gardencorev1beta1.Kubernetes{Version: "1.0.0"}
+
+			// mark latest version 1.02 as preview
+			cloudProfile.Spec.Kubernetes.Versions[3].Classification = &previewClassification
+
+			version, _, err := MaintainKubernetesVersion(shoot, cloudProfile, shootLogger)
+
+			Expect(err).To(BeNil())
```

Comment:
`s/Expect(err).To(BeNil())/Expect(err).NotTo(HaveOccured())/g`
```suggestion
			Expect(err).NotTo(HaveOccured())
```

Reply:
There are still quite a lot of `Expect(err).To(BeNil())` checks, can you replace them with `Expect(err).NotTo(HaveOccurred())`?